geode-core/src/main/java/org/apache/geode/management/internal/ManagementCacheListener.java

```diff
@@ -109,8 +108,15 @@ public void afterUpdate(EntryEvent<String, Object> event) {
 
   }
 
-  void markReady() {
-    readyForEvents = true;
+  private void blockUntilReady() {
+    try {
+      readyForEvents.await();
+    } catch (InterruptedException e) {
+      // ignored
```

Comment:
Anytime you catch an InterruptedException, you should reset the interrupt bit unless the thread is locally owned and it will actually terminate when interrupted:
```
try {
  readyForEvents.await();
} catch (InterruptedException e) {
  Thread.interrupt();
  // recommended: throw new RunnableException(e); -- use an appropriate GemFire exception here
}
```
@upthewaterspout wrote an article but about this but I'm not sure where it is.

Reply:
Also: If you don't rethrow the InterruptedException within a runtime exception, then this code probably needs to perform some sort of loop that checks a CancelCriterion. Grepping 'catch (InterruptedException' on the Geode product code (not tests) should give some good examples.